cmd/sidecars/disk-mutation/diskimage.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Nvidia
+ *
+ */
+
+// Inspired by cmd/example-hook-sidecar
+
+package main
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/pflag"
+
+	vmSchema "kubevirt.io/api/core/v1"
+
+	domainSchema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+const bootDiskImageNameAnnotation = "diskimage.vm.kubevirt.io/bootImageName"
+
+func OnDefineDomain(log *log.Logger, vmiJSON, domainXML []byte) (string, error) {
+	vmiSpec := vmSchema.VirtualMachineInstance{}
+	err := json.Unmarshal(vmiJSON, &vmiSpec)
+	if err != nil {
```

Comment:
Inline check?

Reply:
Again, this is how the code is [at the present](https://github.com/kubevirt/kubevirt/blob/main/cmd/example-disk-mutation-hook-sidecar/diskimage.go#L74), I've only applied the changes related to sidecar-shim plus moving to `cmd/sidecar` folder. I don't mind doing this changes as follow up for example.